symbolic-il2cpp/src/usym.rs

```diff
@@ -144,29 +144,46 @@ mod raw {
     pub(super) struct SourceRecord {
         /// Instruction pointer address, relative to base address of assembly.
         pub(super) address: u64,
-        /// Managed symbol name, as an offset into the strings section.
-        pub(super) symbol: u32,
-        /// Managed source file, as an offset into the strings section.
-        pub(super) file: u32,
-        /// Managed line number.
-        pub(super) line: u32,
-        // These might not even be u64, it's just 128 bits we don't know.
-        _unknown0: u64,
-        _unknown1: u64,
+        /// Native symbol name, as an offset into the strings section.
+        pub(super) native_symbol: u32,
+        /// Native source file, as an offset into the strings section.
+        pub(super) native_file: u32,
+        /// Native line number.
+        pub(super) native_line: u32,
+        /// Managed code symbol name, an offset into the strings section.
+        ///
+        /// 0 if code does not map to managed code.
```

Comment:
i changed this line as well as its equivalent in managed file so instead of saying "empty string if code.." it says 0.  i think it makes it more clear exactly what you should expect or check for if you're working directly with the raw SourceRecord.

Reply:
I think you might be making an assumption too many here: The offset of the empty string just happens to be at offset 0 in the strings section in our example.  This does not mean that it has to be.  The fact that an empty string is part of the strings section at all to me means we should use empty string here and not offset 0.

Reply:
sorry, it looks like i misunderstood the original comment, then. as far as i can tell from fiddling around and generating a few different usyms, it looks like the empty string is consistently set as the first element in the string table. 

i think unless there's any evidence to prove otherwise, it should be fine to assume that these fields being set to 0 (or really, any offset into the string table) effectively means "nothing" until we're given any reason to believe otherwise. of course, this doesn't mean that only empty strings will have an offset of 0, but if we do find an offset of 0 it's incredibly likely to be empty.